tools/sgcspevaluatorcli/csp.go

```diff
@@ -0,0 +1,77 @@
+package sgcspevaluatorcli
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// ContentSecurityPolicy is a map where each key is a Content Security Policy directive,
+// and value is a list of sources for the respective directive.
+type ContentSecurityPolicy map[string][]string
+
+// String returns ContentSecurityPolicy ready to be send as a HTTP header.
+func (csp ContentSecurityPolicy) String() string {
+	directives := make([]string, 0, len(csp))
+	for directive, rules := range csp {
+		sort.Slice(rules, func(i, j int) bool { return rules[i] < rules[j] })
+		directives = append(directives, fmt.Sprintf("%s %s", directive, strings.Join(rules, " ")))
+	}
+	sort.Slice(directives, func(i, j int) bool { return directives[i] < directives[j] })
+	return strings.Join(directives, "; ")
+}
+
+// Clone returns a deep copy of the content security policy.
+func (csp ContentSecurityPolicy) Clone() ContentSecurityPolicy {
+	cloned := map[string][]string{}
+	for key, values := range csp {
+		cloned[key] = make([]string, len(values))
+		copy(cloned[key], values)
+	}
+	return cloned
+}
+
+// With returns a new content security policy with an additional directive and sources.
+func (csp ContentSecurityPolicy) With(directive, source string, sources ...string) ContentSecurityPolicy {
```

Comment:
how come we separate out the first vararg to a separate arg here?

Reply:
that's to force at least one source on the compilation level.

i.e.
```
csp.With("directive")
```

won't compile

Reply:
ah! I see, nice way of avoiding errors/panics!